crd-generator/apt/src/main/java/io/fabric8/crd/generator/apt/CustomResourceAnnotationProcessor.java

```diff
@@ -160,4 +161,34 @@ public void close() throws IOException {
       out.close();
     }
   }
+
+  private static Optional<ClassRef> getCustomResourceRef(TypeDef typeDef) {
+    return superClassStream(typeDef)
+      .filter(s -> s.getDefinition().getFullyQualifiedName().equals(CustomResource.class.getName()))
+      .findFirst();
+  }
+
+  private static Optional<ClassRef> getCustomResourceSpec(TypeDef typeDef) {
+    return getCustomResourceTypeParam(typeDef, 0);
+  }
+
+  private static Optional<ClassRef> getCustomResourceStatus(TypeDef typeDef) {
+    return getCustomResourceTypeParam(typeDef, 1);
+  }
+
+  private static Optional<ClassRef> getCustomResourceTypeParam(TypeDef typeDef, int paremeterIndex) {
```

Comment:
I would even move the custom resource ref resolution outside to avoid walking the hieararchy, which is a change I tried to push to the previous PR.